storage/flash_ll.h

```diff
@@ -0,0 +1,70 @@
+/*
+ * This file is part of the Trezor project, https://trezor.io/
+ *
+ * Copyright (c) SatoshiLabs
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef FLASH_LL_H_
+#define FLASH_LL_H_
+
+#include <secbool.h>
+#include <stdint.h>
+
+// Flash memory low-level API, providing abstraction for
+// various flash archictures found on STM32 MCUs
+
+// The 'sector' parameter in this API can represent
+//    1. Non-uniform sector number on STM32F4
+//    2. Uniform page number on STM32U5
+
+// Returns the size of the a continuous area of sectors
+// Returns 0 if any of the sectors is out of range
+uint32_t flash_sector_size(uint16_t first_sector, uint16_t sector_count);
+
+// Returns number of the sector/page at specified byte 'offset'
+// from the beginning of the 'first_sector'
+uint16_t flash_sector_find(uint16_t first_sector, uint32_t offset);
+
+// Returns the physical address of a byte on specified 'offset' in the specified
+// 'sector'. Checks if it's possible to access continues space of 'size' bytes
+// Returns NULL i [offset, offset + size] is of out of the specified sector
+const void *flash_get_address(uint16_t sector, uint32_t offset, uint32_t size);
+
+// Unlocks the flash memory for writes/erase operations
+// Flash must be locked again as soon as possible
+secbool __wur flash_unlock_write(void);
+
+// Locks the flash memory for writes/erase operations
+secbool __wur flash_lock_write(void);
+
+// Writes a single byte to the specified 'offset' inside a flash 'sector'
+secbool __wur flash_write_byte(uint16_t sector, uint32_t offset, uint8_t data);
```

Comment:
only when FLASH_BYTE_ACCESS is set. (also applies to write word below)

Reply:
fixed, see fcc8ab6cb635bbf92c33d57df3e5953a656e9be6